### PR TITLE
Added delegate function call after quorum #286

### PR DESCRIPTION
**Pull request by dechjo**

### CODE REVIEW

**Comment by vshvsh** [comment]
"Delegate call" has a specific meaning in eth, let's not use this word

File: contracts/0.4.24/oracle/LidoOracle.sol [hunk 1]
```diff
@@ -152,6 +155,10 @@ contract LidoOracle is ILidoOracle, AragonApp {
         _tryPush();
     }
 
+    function setQuorumDelegate(address _addr, bytes4 _sig) external /* TODO: auth? */ {
```
Comment (vshvsh) [inline contracts/0.4.24/oracle/LidoOracle.sol:158]: only by voting

File: contracts/0.4.24/oracle/LidoOracle.sol [hunk 2]
```diff
@@ -414,6 +421,16 @@ contract LidoOracle is ILidoOracle, AragonApp {
         POST_COMPLETED_TOTAL_POOLED_ETHER_POSITION.setStorageUint256(ISTETH(lido).totalSupply());
         PREV_COMPLETED_EPOCH_ID_POSITION.setStorageUint256(LAST_COMPLETED_EPOCH_ID_POSITION.getStorageUint256());
         LAST_COMPLETED_EPOCH_ID_POSITION.setStorageUint256(epochId);
+
+        // call the quorum delegate
+        uint256 quorumDelegateRaw = QUORUM_DELEGATE_POSITION.getStorageUint256();
+        (uint256 lastTotalPooledEther,
+         uint256 prevTotalPooledEther,
+         uint256 timeElapsed) = getLastCompletedReports();
+        address(quorumDelegateRaw).call(bytes4(quorumDelegateRaw >> 160),
```
Comment (vshvsh) [inline contracts/0.4.24/oracle/LidoOracle.sol:430]: if delegate is not 0

File: contracts/0.4.24/oracle/LidoOracle.sol [hunk 3]
```diff
@@ -86,6 +86,9 @@ contract LidoOracle is ILidoOracle, AragonApp {
     bytes32 internal constant LAST_COMPLETED_EPOCH_ID_POSITION = keccak256("lido.LidoOracle.lastCompletedEpochId");
     bytes32 internal constant PREV_COMPLETED_EPOCH_ID_POSITION = keccak256("lido.LidoOracle.prevCompletedEpochId");
 
+    /// @dev function credentials to be called when the quorum is reached
+    bytes32 internal constant QUORUM_DELEGATE_POSITION = keccak256("lido.LidoOracle.quorumDelegate");
```
Comment (vshvsh) [inline contracts/0.4.24/oracle/LidoOracle.sol:90]: "Delegate call" has a specific meaning in eth, let's not use this word

File: contracts/0.4.24/oracle/LidoOracle.sol (hunk 2 above)
Comment (vshvsh) [reply contracts/0.4.24/oracle/LidoOracle.sol:430]: it can be 0

File: contracts/0.4.24/oracle/LidoOracle.sol (hunk 1 above)
Comment (vshvsh) [reply contracts/0.4.24/oracle/LidoOracle.sol:158]: we can require function sig to be constant (e.g. "processLidoOracleReport") to simplify the code and interfaces

**Review by vshvsh — CHANGES_REQUESTED** [review]

File: contracts/0.4.24/oracle/LidoOracle.sol (hunk 2 above)
Comment (dechjo) [reply contracts/0.4.24/oracle/LidoOracle.sol:430]: it is OK: in this case the call simply returns `false` (which we ignore anyway)

File: contracts/0.4.24/oracle/LidoOracle.sol (hunk 1 above)
Comment (dechjo) [reply contracts/0.4.24/oracle/LidoOracle.sol:158]: yes, sounds good, will do

File: contracts/0.4.24/oracle/LidoOracle.sol (hunk 3 above)
Comment (dechjo) [reply contracts/0.4.24/oracle/LidoOracle.sol:90]: Renaming to `QUORUM_CALLBACK`...

File: contracts/0.4.24/oracle/LidoOracle.sol (hunk 2 above)
Comment (dechjo) [reply contracts/0.4.24/oracle/LidoOracle.sol]: moved to using interface `IQuorumCallback` and thus added the check.

Comment (vshvsh) [reply contracts/0.4.24/oracle/LidoOracle.sol]: I mean that callback can be simply not set. If you call `0x0.processLidoOracleReport(...)` we'll revert.

Comment (dechjo) [reply contracts/0.4.24/oracle/LidoOracle.sol]: Yes, now did so. And tested.
I meant that previously when I did `address(callback).call(bytes4(keccak256("...")), ...)`, this construction would not revert even if the `callback` was 0.

Comment (vshvsh) [reply contracts/0.4.24/oracle/LidoOracle.sol]: huh, right. I assumed calling an EOA would revert.

**Pull request merged**